docs/auth-arch.md

```diff
@@ -0,0 +1,69 @@
+# Authentication architecture
+
+**Glossary:**
+
+- **CL** - client
+- **BE** - business logic (backend)
+- **JMS** - Jellyfish Media Server
+
+## Approaches and connection with signaling architecture
+
+Authentication might be a big factor when deciding on signaling architecture (see the other document). We need to consider 2 situations:
+
+- **CL** connect directly do **JMS**,
+- **CL** connect to **BE** which forwards signaling messages to **JMS**.
+
+Let's start with the first approach.
+
+### Direct connection between **CL** and **JMS**
+
+Let's assume client wants to join some multimedia room and is already authenticated (from the bussines logic perspective, e.g. he is logged in to his account).
+Also, **JMS** and **BE** share common secret (more about that later).
+Scenario:
+
+1) **CL** sends request to join the multimedia room to **BE**.
+2) **BE** recieves the request and sends `add_peer` request to **JMS**.
+3) **JMS** sends back created `peer_id`.
+4) **BE** uses received `peer_id`, id of the **JMS** instance to create JWT token with that is signed with the secret (or the **JMS** creates the token in the previous step).
+The token can also contain permissions (which may differ between "normal" client and administrator).
+5) **BE** responds to **CL** with the token.
+6) **CL** can now open WebSocket connection with **JMS** using the token to authorize. **JMS** (thanks to informations included in the token)
+can tell who opened the connection and that it was intended for this instance of **JMS**.
+
+Problems:
+
+- who creates the token? (more on that later),
+- the token can expire (this doesn's seem like a problem, when the token is used only to open WebSocker connection, but user possibly needs to implement logic to refresh the token).
+
+### Using **BE** as a middleman
+
+The same assumptions and first 3 steps as in the previous example:
+
+4) **BE** responds to **CL** that it was authorized and that it can send signaling messages to **BE**.
+
+Problems:
+
+- in current implementation of `membrane_rtc_engine`, endpoint should be created after signaling connection was established. In previous example it was obvious:
```

Comment:
> in current implementation of `membrane_rtc_engine`, endpoint should be created after signaling connection was established
Yes, it would probably be a good idea, but preamptively creating an endpoint would be acceptable if you made sure to clean it up - the endpoint awaits `join` media event before becoming fully active.